Source/EasyNetQ/PersistentConnection.cs

```diff
@@ -99,39 +99,38 @@ void TryToConnect(object timer)
                         break;
                     }
 
-                    connectionFactory.Success();
+                    try
+                    {
+                        OnConnected();
+                    }
+                    catch
+                    {
+                        connection.Dispose();
+                        throw;
+                    }
                     
+                    connection.ConnectionShutdown += OnConnectionShutdown;
```

Comment:
What is the reason of changing the order of calling `OnConnected` and subscribing to connection events?

Reply:
This change is modifying the workflow to only consider the connection successfully opened after OnConnected has opened the channel. If we have to retry with a new connection, it seems unhelpful to fire off irrelevant events for a connection that will just be discarded anyway.